pkg/parquet/reader.go

```diff
@@ -206,6 +221,87 @@ func (r *s3Reader) ReadFileIntoTarget(ctx context.Context, file string, target i
 	return nil
 }
 
+func (r *s3Reader) ReadFileColumns(ctx context.Context, columnNames []string, file string, batchSize int, offset int64) (ReadResults, error) {
+	columnNameSet := make(map[string]struct{})
```

Comment:
I think we should make this the default behaviour, otherwise we only support this for a few use cases which is not really consistent (and leads to quite some almost-duplicates)